### PR TITLE
{AKS} add static egress gateway related cli

### PR DESCRIPTION
**Pull request by jwtty**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
```
az aks create <CLUSTER_NAME> --enable-static-egress-gateway
az aks update <CLUSTER_NAME> --enable-static-egress-gateway
az aks update <CLUSTER_NAME> --disable-static-egress-gateway
az aks nodepool add <NODEPOOL_NAME> --mode Gateway --gateway-prefix-size 31
```

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by FumingZhang — COMMENTED** [review]
lgtm

Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=91865855&view=results), test passed
- test_aks_and_agentpool_with_static_egress_gateway

**Review by FumingZhang — APPROVED** [review]

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -22,6 +22,13 @@ Pending
 +++++++
 * Ignore invalid ip error for `--api-server-authorized-ip-ranges`.
 
+3.0.0b8
++++++++
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:26]: should move these to pending or choose a new version

Comment (jwtty) [reply src/aks-preview/HISTORY.rst]: thanks, fixed

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by jwtty** [comment]
> Please fix CI issues

fixed, it was due to not up-to-date recording file.

**Review by chasewilson — APPROVED** [review]

**Review by Moazzem-Hossain-pixel — APPROVED** [review]
Gladly approved.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**